### DISCUSSION TITLE
WebQuery enhancement with%Varx% substitution

### DISCUSSION THREAD

**Discussion by pierpaolopalazzo** [arendst/Tasmota#16998] [Ideas]
I desire a little enhancement. 

I need variable replacement in WebQuery command. 

What I need is to have something like:

ON Switch#toggle DO Backlog WebQuery http://192.168.1.100/on.php?power=%Var1% GET ENDON

**Comment by barbudor** [comment]
That should already be working like this 
In rules Every %varx% is replaced by the value before execution

**Reply by pierpaolopalazzo** [reply]
Wow. I tried on web console and it doesn't do var substitution. I try in rules then.

**Comment by barbudor** [comment]
Indeed variable substitution works only in rule, not in console
